nexus/src/app/background/region_replacement.rs

```diff
@@ -0,0 +1,61 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Background task for detecting regions that need replacing and beginning that
+//! process
+//!
+//! TODO this is currently a placeholder for a future PR
+
+use super::common::BackgroundTask;
+use crate::app::sagas::SagaRequest;
+use futures::future::BoxFuture;
+use futures::FutureExt;
+use nexus_db_queries::context::OpContext;
+use nexus_db_queries::db::DataStore;
+use serde_json::json;
+use std::sync::Arc;
+use tokio::sync::mpsc::Sender;
+
+pub struct RegionReplacementDetector {
+    _datastore: Arc<DataStore>,
+    _saga_request: Sender<SagaRequest>,
+}
+
+impl RegionReplacementDetector {
+    pub fn new(
+        datastore: Arc<DataStore>,
+        saga_request: Sender<SagaRequest>,
+    ) -> Self {
+        RegionReplacementDetector {
+            _datastore: datastore,
+            _saga_request: saga_request,
+        }
```

Comment:
It's not clear to me why we're including the stub task for this `RegionReplacementDetector` - you mentioned in chat that this task is necessary, but we aren't even doing anything with this channel, and we **can't** do anything with it, because it only contains a test-only variant?

Reply:
To be clear, it's necessary to put the `Sender<SagaRequest>` somewhere so that it's not dropped, as that would cause Nexus to panic (with the changes in this PR).

Reply:
I understand that those are the mechanics of monitoring a channel, but from my perspective, we're plumbing a channel that we can **never** use in production because it only contains a test-only variant. I'm just not sure what I'm reviewing -- what is this channel **supposed** to be passing around? Why is it necessary?